### PR TITLE
waybar: lock the wifi-signal-strength script

### PR DESCRIPTION
**Pull request by Nerox9**
## Description of changes

Multiple call from multiple instance of waybar causes race condition. flock and trap are used to eliminate. 

## Checklist for things done

- [x] Summary of the proposed changes in the PR description
- [x] More detailed description in the commit message(s)
- [x] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [ ] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [ ] Test procedure described (or includes tests). Select one or more:
  - [x] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [ ] Author has run `nix flake check --accept-flake-config` and it passes
- [ ] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [ ] Author has added reviewers and removed PR draft status

### CODE REVIEW

File: packages/wifi-signal-strength/default.nix [hunk 1]
```diff
@@ -13,8 +13,12 @@
     pkgs.writeShellScript
     "wifi-signal-strength"
     ''
+      trap "${pkgs.openssh}/bin/ssh -q -S /tmp/nmcli_socket -O exit ghaf@${netvm_address}" EXIT
```
Comment (brianmcgillion) [inline packages/wifi-signal-strength/default.nix:16]: https://github.com/tiiuae/ghaf/pull/549/files#diff-0dd9147b12941844c66d8519a708fb64e934748abb257ff28c09de8bec9dc667R47

Do you not already do that here? 

If I am not mistaken EXIT only triggers on normal exit (0), which only occurs after the line listed above. So could we replace line 47 with an exit 0 to trigger the trap?

**Review by brianmcgillion — COMMENTED** [review]
Looking good, just the one question on the trap

File: packages/wifi-signal-strength/default.nix (hunk 1 above)
Comment (Nerox9) [reply packages/wifi-signal-strength/default.nix:16]: Yes, you are right. The script will be clearer with it, thanks!

**Review by brianmcgillion — APPROVED** [review]

**Comment by milva-unikie** [comment]
Tested with X1 that this does fix the problem and everything else is working normally.

**Pull request merged**